pkg/controller/job/job_controller.go

```diff
@@ -944,10 +957,14 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 			}
 			jobCtx.podsWithDelayedDeletionPerIndex = getPodsWithDelayedDeletionPerIndex(logger, jobCtx)
 		}
-		if jobCtx.finishedCondition == nil {
-			if msg, met := matchSuccessPolicy(logger, job.Spec.SuccessPolicy, *job.Spec.Completions, jobCtx.succeededIndexes); met {
-				jobCtx.finishedCondition = newCondition(batch.JobSuccessCriteriaMet, v1.ConditionTrue, batch.JobReasonSuccessPolicy, msg, jm.clock.Now())
-			}
```

Comment:
I'm on the fence whether or not I should revert this change. But when reverting this, 2.3 and 3 comments might look like mixed.

Any thoughts?

Reply:
TBH, I would avoid changing the production code just for the  comments.

Reply:
I'm with Michal here, let's leave it as is.

Reply:
That makes sense. Let me revert this

Reply:
Done.